tests/python/test_fields_builder.py

```diff
@@ -6,171 +6,177 @@
 
 @ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan, ti.metal])
 def test_fields_with_shape():
-    n = 5
-    x = ti.field(ti.f32, [n])
+    shape = 5
+    x = ti.field(ti.f32, shape=shape)
+    y = ti.field(ti.f32, shape=shape)
 
     @ti.kernel
-    def func():
-        for i in range(n):
+    def assign_field_single():
+        for i in range(shape):
             x[i] = i
 
-    func()
-
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i
 
-    y = ti.field(ti.f32, [n])
-
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple():
+        for i in range(shape):
             y[i] = i * 2
-        for i in range(n):
+        for i in range(shape):
             x[i] = i * 3
 
-    func2()
-
-    for i in range(n):
+    assign_field_multiple()
+    for i in range(shape):
         assert x[i] == i * 3
         assert y[i] == i * 2
 
-    func()
-
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i
 
 
 @ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan, ti.metal])
 def test_fields_builder_dense():
-    n = 5
-
+    shape = 5
     fb1 = ti.FieldsBuilder()
     x = ti.field(ti.f32)
-    fb1.dense(ti.i, n).place(x)
+    fb1.dense(ti.i, shape).place(x)
     fb1.finalize()
 
     @ti.kernel
-    def func1():
-        for i in range(n):
+    def assign_field_single():
+        for i in range(shape):
             x[i] = i * 3
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
     fb2 = ti.FieldsBuilder()
     y = ti.field(ti.f32)
-    fb2.dense(ti.i, n).place(y)
+    fb2.dense(ti.i, shape).place(y)
     z = ti.field(ti.f32)
-    fb2.dense(ti.i, n).place(z)
+    fb2.dense(ti.i, shape).place(z)
     fb2.finalize()
 
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_field_multiple():
+        for i in range(shape):
             x[i] = i * 2
-        for i in range(n):
+        for i in range(shape):
             y[i] = i + 5
-        for i in range(n):
+        for i in range(shape):
             z[i] = i + 10
 
-    func2()
-    for i in range(n):
+    assign_field_multiple()
+    for i in range(shape):
         assert x[i] == i * 2
         assert y[i] == i + 5
         assert z[i] == i + 10
 
-    func1()
-    for i in range(n):
+    assign_field_single()
+    for i in range(shape):
         assert x[i] == i * 3
 
 
-@ti.test(arch=[ti.cpu, ti.cuda, ti.metal])
+@ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan, ti.metal])
```

Comment:
```suggestion
@ti.test(arch=[ti.cpu, ti.cuda, ti.metal])
```
You can see CI fails. This is because the Vulkan backend doesn't support `pointer` yet.

Reply:
Yeah. Fixed.